pageserver/src/page_service.rs

```diff
@@ -582,21 +541,22 @@ impl PageServerHandler {
         info!("flushing layers");
         timeline.checkpoint(CheckpointConfig::Flush)?;
 
+        timeline.launch_wal_receiver()?;
```

Comment:
I think, timeline creation could be extracted so that there's only one `pub` method that does all the magic, creating an empty timeline and initializing it.
This way, nobody but the timeline itself needs to know about `timeline.launch_wal_receiver`.

I've approached this in my not-finished branch about atomic timelines (see `initialize_new_timeline` and around): https://github.com/neondatabase/neon/compare/kb/tenant-state?expand=1#diff-2dee9987054422607c1ab7369e102927d8477dea6d75c4c7f3891e3d65b15d1eR318

Reply:
Agreed. Let's do that in a separate PR, though.